vector/src/main/java/im/vector/app/features/call/webrtc/WebRtcCall.kt

```diff
@@ -770,12 +775,37 @@ class WebRtcCall(
         return currentCaptureFormat
     }
 
-    fun startSharingScreen() {
-        // TODO. Will be handled within the next PR.
+    fun startSharingScreen(videoCapturer: VideoCapturer) {
+        val factory = peerConnectionFactoryProvider.get() ?: return
+
+        this.videoCapturer = videoCapturer
+
+        val localMediaStream = factory.createLocalMediaStream(STREAM_ID)
+        val videoSource = factory.createVideoSource(videoCapturer.isScreencast)
+
+        // Start capturing screen
+        val surfaceTextureHelper = SurfaceTextureHelper.create("CaptureThread", rootEglBase!!.eglBaseContext)
+        videoCapturer.initialize(surfaceTextureHelper, context, videoSource.capturerObserver)
+        videoCapturer.startCapture(currentCaptureFormat.width, currentCaptureFormat.height, currentCaptureFormat.fps)
+
+        // Remove local camera previews
+        localSurfaceRenderers.forEach { it.get()?.let { localVideoTrack?.removeSink(it) } }
+
+        // Show screen preview locally
+        localVideoTrack = factory.createVideoTrack(VIDEO_TRACK_ID, videoSource).apply { setEnabled(true) }
+        localMediaStream?.addTrack(localVideoTrack)
+        localSurfaceRenderers.forEach { it.get()?.let { localVideoTrack?.addSink(it) } }
+
+        // Remove camera stream
+        peerConnection?.removeTrack(videoSender)
+
+        screenSender = peerConnection?.addTrack(localVideoTrack, listOf(STREAM_ID))
```

Comment:
Each of these "regions" (as separated by the comments) can be made more readable by each being their own private function. This method can then read like a story without any comments

Reply:
Done. This class needs refactoring.